nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-api/src/main/java/org/apache/nifi/oauth2/OAuth2AccessTokenProvider.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.oauth2;
+
+import org.apache.nifi.controller.ControllerService;
+
+/**
+ * Controller service that provides OAuth2 tokens
+ */
+public interface OAuth2AccessTokenProvider extends ControllerService {
+
+    /**
+     * @return A valid OAuth2 access token
+     */
+    String getAccessToken();
```

Comment:
Is there a particular reason for changing the interface to return a `String` as opposed to an object of some kind?

Reply:
This is a published API, so I'm not going to back this change without a good reason.

Reply:
I thought simply accessing the token itself is enough but I guess it makes sense to be able to access the corresponding metadata. I'll change it to return `AccessDetails`.